app/models/user.rb

```diff
@@ -159,9 +155,21 @@ def with_interest(interest)
       where(":interest = ANY(interested_in)", interest: interest)
     end
 
+    def with_location(location)
+      where(country: location)
+    end
+
+    def as_coach_availability
+      where(availability: true)
+    end
+
     def immutable_attributes
       [:github_handle]
     end
+
+    def reset_availability!
```

Comment:
Can you add tests for this? It's a method with rather large impact... Would be good to know if it's really (and only) doing what it should 😉